### DISCUSSION TITLE
check if the detection is correct or not

### DISCUSSION THREAD

**Discussion by jehan88** [ultralytics/yolov5#8836] [Q&A]
dear All,
how can I check if the detection is correct or not?
I mean if the object  with the same label or not ,like its door but labeled as window or door?

**Comment by glenn-jocher** [comment]
@jehan88 val.py compares detections against labels

**Reply by jehan88** [reply]
thank you for the answer, but val.py with which weight? does I need to use the best weight from train.py?

**Reply by glenn-jocher** [reply]
@jehan88 see val.py Usage examples at beginning of file, the idea is you pass your own --weights and your own --data
https://github.com/ultralytics/yolov5/blob/2e10909905b1e0e7eb7bac086600fe7ee2c0e6a5/val.py#L2-L18
